### PR TITLE
instructorStudentComment: Give more informative status messages #1426

### LINKED ISSUE #1426: instructorStudentComment: Give more informative status messages
_From [dam...@gmail.com](https://code.google.com/u/114919422028123491955/) on January 04, 2014 11:26:04_

Instead of generic messages, be more informative. e.g. The comment was deleted, also give an extract of the comment "The comment "blah blah" was deleted".

_Original issue: http://code.google.com/p/teammatespes/issues/detail?id=1427_

### PR DESCRIPTION
**Pull request by samsontmr**
Fixes #1426  

**Problem**
Comment status messages too generic.

**Solution**
Use String.format to add comment excerpt into status messages.

### CODE REVIEW

**Comment by whipermr5** [comment]
After tracing, I found out that the `<p>` tags are added by the UI (HTML comment editor) to the `commenttext` form data itself when the comment is submitted:
screen shot 2017-01-19 at 3 09 00 pm

Since this is a UI browser test, the comment (in the screenshot above, `test2`) goes through the editor, the `<p>` tags are added, the backend receives `<p>test2</p>` as the comment, and it gets reflected in the status message of the resulting page, as expected. To get the actual value, the test calls `verifyStatus`, which uses `getText` on the `WebElement`; this renders the `<p>` tags and results in a newline before the comment. Therefore, all is well - you should just include a newline before the comment in the expected status due to the presence of the HTML editor in UI testing.

**Comment by samsontmr** [comment]
@whipermr5 Wow that's amazing, how did you manage to trace it? Thanks!!

**Comment by whipermr5** [comment]
For the record:
screenshot_20170120-004013 01

**Comment by samsontmr** [comment]
Hm adding a System.lineSeparator() doesn't seem to work either. I'll need to investigate further

**Comment by whipermr5** [comment]
Maybe check out Selenium's `WebElement.getText()` method to see what character they use to separate paragraphs

**Comment by whipermr5** [comment]
To make it clearer: the comment is stored as `<p>test2</p>` from input all the way to storage and display. The only time it gets converted to a newline is within the test, when Selenium is used to retrieve the text from the HTML element in the actual output displayed by the browser.

**Comment by whipermr5** [comment]
A cleaner way might be to use the HTML editor's source code feature to input plaintext comments within the test code that adds the comment, to avoid HTML tags getting added.

**Comment by samsontmr** [comment]
You mean the main code right? Since the tag is coming from actual output.

**Comment by whipermr5** [comment]
The tag is coming from the UI, driven by the test code. Take a look at `[addComment](https://github.com/TEAMMATES/teammates/blob/master/src/test/java/teammates/test/pageobjects/InstructorStudentRecordsPage.java#L43)` and `[submitCommentToCourse](https://github.com/TEAMMATES/teammates/blob/7a2eb84eda7105548d7d13cdaf5065778f7468f0/src/test/java/teammates/test/pageobjects/InstructorCourseDetailsPage.java#L87)`

**Comment by samsontmr** [comment]
Ah so tags are only introduced during testing and not present during actual usage

**Comment by whipermr5** [comment]
I imagine in actual usage, all the comments have HTML tags added unless the user specifically chooses to enter plaintext by clicking on the source code button of the HTML editor

**Comment by samsontmr** [comment]
Yea so that means we would have to deal with the tags in the main code and not the testig code

**Comment by whipermr5** [comment]
I feel the tags are fine and should be left there. To avoid displaying them in the informative status messages, I suggest pre-processing the comments before displaying them (run the comment through an HTML parser, extract the plaintext and pick the first n characters to display). Right now, your changes make it possible for people to inject javascript into pages by submitting comments with `<script>` tags. This is dangerous as it can be exploited to perform reflected (and persistent) XSS attacks.

**Comment by samsontmr** [comment]
Ready for Review

File: src/main/java/teammates/ui/controller/InstructorStudentCommentAddAction.java [hunk 1]
```diff
@@ -59,9 +62,9 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         try {
             CommentAttributes createdComment = logic.createComment(comment);
             //TODO: move putDocument to Task Queue
-            logic.putDocument(createdComment);
```
Comment (wkurniawan07) [inline src/main/java/teammates/ui/controller/InstructorStudentCommentAddAction.java]: Do not remove this line; it's used for enabling search to be performed on student comments.

Comment (samsontmr) [inline src/main/java/teammates/ui/controller/InstructorStudentCommentAddAction.java]: thanks for pointing that out!

**Comment by damithc** [comment]
@samsontmr for future reference, let's not put 'WIP' in issue title as it gets stuck as the subject of notification email thread even after it is removed from the title. Instead, mention WIP in the PR description.

**Comment by samsontmr** [comment]
@damitch noted

**Pull request merged**